apps/backend/agents/core/tutoring_agent.py

```
@@ -0,0 +1,358 @@
+Student Question/Problem: {question}
+
+Grade Level: {grade_level}
+Subject: {subject}
+
+Student Context:
+- Current Understanding: {student_context.get('understanding_level', 'Unknown')}
+- Learning Style: {student_context.get('learning_style', 'Visual')}
+"""
+        
+        if previous_attempts:
+            input_text += f"\nPrevious Attempts:\n"
```

Ruff (0.14.4)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)